typed-racket-lib/typed-racket/typecheck/tc-envops.rkt

```diff
@@ -39,6 +40,9 @@
             [(cons p ps)
              (match p
                [(TypeProp: (Path: pes (? identifier? x)) pt)
+                #:when (match pes
```

Comment:
This (and line 79) should be a helper function or match expander --- there's going to be more cases like `VecLenPE` in the future, right?

Reply:
Good point -- I just refactored this part quite a bit and created an `uninterpreted-PE?` predicate that can later be extended to include similar path elements w/o modifying this code.

Now to test the refactor...